### ISSUE TITLE
[bug]

### ISSUE THREAD

**Issue by Coober-Ding** [Rich-Harris/magic-string#149]
example code:

```js
const MagicString = require('../dist/magic-string.umd.js')
const SourceMapConsumer = require( 'source-map' ).SourceMapConsumer;

const s = '0123456789';
const ms = new MagicString(s);

//remove '67' from s
const ns = ms.remove(6,8).toString();

var map = ms.generateMap({hires: true});
var smc = new SourceMapConsumer(map);

console.log(`${ns} = ${s}`);
for (var i = 0;i < ns.length; i++) {
    loc = smc.originalPositionFor({ line: 1, column: i });
    console.log(`${ns[i]} = ${s[loc.column]}`);
}
```

console:
```
01234589 = 0123456789
0 = 0
1 = 1
2 = 2
3 = 3
4 = 4
5 = 5
8 = 6 //is this right???
9 = 9
```

/src/utils/Mappings.js
row:50
`this.pending = [this.generatedCodeColumn, sourceIndex, loc.line, loc.column];`
row:19
`this.rawSegments.push(this.pending);`

**Comment by M4rk9696** [comment]
@mourner Is this issue still present?

Thought I'll have a look. Could not reproduce with 

magic-string: 0.25.3
source-map: 0.7.3

**Comment by kzc** [comment]
This bug still exists on master branch a312519cfe9caa78ade7f09cc2b07459d3d17f4d.

Fixed by PR #159.
